### PR TITLE
Reduce 6.0 release branch CI runs (6.0 preview 2)

### PR DESCRIPTION
**Pull request by BruceForstall**
6.0 release branches are running AzDO stress pipelines on a schedule, even though (normally) nothing is changing. Change this to only run if something changes.

This is a port of https://github.com/dotnet/runtime/pull/49020 to the release branch.

### CODE REVIEW

File: eng/pipelines/coreclr/jitstress.yml
Comment (hoyosjs) [inline eng/pipelines/coreclr/jitstress.yml:8]: Master no longer exists, I'd revert these lines of the change as this will only after the branch you check it in.

**Review by hoyosjs — APPROVED** [review]
Just a small nit.

**Comment by mmitche** [comment]
We should hold off on this. I believe the last runtime change for p2 has left the station

File: eng/pipelines/coreclr/jitstress.yml
Comment (BruceForstall) [reply eng/pipelines/coreclr/jitstress.yml:8]: I suppose we could just delete the "master" section of this (and other) yml files, since they're unused in the release branches anyway.

I did this just as a "cherry-pick" from the 'main' branch change, so it doesn't really hurt, either

**Comment by BruceForstall** [comment]
> We should hold off on this. I believe the last runtime change for p2 has left the station

@mmitche the issue here is that this is an infra-only change. The 6.0p1 and 6.0p2 branches continue to run these pipelines 4 times per week, and I want them to stop (unless needed).

**Comment by mmitche** [comment]
Gotcha. Even if there are no changes?

Anyways, please cancel the official build once this is merged.

**Comment by BruceForstall** [comment]
e.g., look at all the "6.0-preview1" and "6.0-preview2" runs here: https://dev.azure.com/dnceng/public/_build?definitionId=658

**Comment by BruceForstall** [comment]
Yes, the pipelines are running even if there are no changes (now).

> Anyways, please cancel the official build once this is merged.

You'll need to tell me how that is done. (does the official build automatically get kicked off for every merge?)

**Review by ViktorHofer — APPROVED** [review]

**Comment by BruceForstall** [comment]
@ViktorHofer @hoyosjs @safern This change (and the related 6.0 preview 1 change: https://github.com/dotnet/runtime/pull/49168) didn't work. I change the YAML to use "always: false" for the schedule, meaning it should only trigger if there is a change. However, we're still getting test runs every day, and they all have a single change. Technically, the documentation says, "if there have been source code changes since the last successful scheduled run". Since there hasn't been a successful run, that's probably why we're getting runs.

Given that we don't care about these runs, I propose to push new PRs to remove the schedules entirely for the preview releases, as well as for main (before it gets branched again).

Comments?

**Pull request merged**
